integtest/src/main/java/org/corfudb/universe/cluster/docker/CorfuServerDockerized.java

```diff
@@ -0,0 +1,233 @@
+            hostPorts.add(PortBinding.of("0.0.0.0", port));
+            portBindings.put(port, hostPorts);
+        }
+
+        HostConfig hostConfig = HostConfig.builder()
+                .portBindings(portBindings)
+                .autoRemove(true)
+                .build();
+
+        // Compose command line for starting Corfu
+        String cmdLine = String.format(
+                "java -cp *.jar org.corfudb.infrastructure.CorfuServer %s",
+                getCommandLineParams()
+        );
+
+        try {
+            return ContainerConfig.builder()
+                    .hostConfig(hostConfig)
+                    .image("corfu-server:" + getAppVersion())
+                    .hostname(params.getGenericName())
+                    .exposedPorts(ports)
+                    .cmd("sh", "-c", cmdLine)
+                    .build();
+        } catch (XmlPullParserException | IOException e) {
+            throw new NodeException("Can't parse application version", e);
+        }
+    }
+
+    private void addShutdownHook(){
+        //Just in case if a test failed and didn't kill the container
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+            try {
+                kill();
+            } catch (NodeException ex) {
+                log.debug("Can't kill container. Already killed. Id: {}", containerId);
+            }
+        }));
+    }
+
+    private String getAppVersion() throws IOException, XmlPullParserException {
+        MavenXpp3Reader reader = new MavenXpp3Reader();
+        Model model = reader.read(new FileReader("pom.xml"));
+        return model.getParent().getVersion();
+    }
+
+    /**
+     * This method create a command line string for starting Corfu server
+     *
+     * @return command line parameters
+     */
+    private String getCommandLineParams() {
+        StringBuilder cmd = new StringBuilder();
+        cmd.append("-a ").append("0.0.0.0");
+
+        switch (params.getPersistence()){
+            case DISK:
+                if(StringUtils.isEmpty(params.getLogDir())){
+                    throw new ClusterException("Invalid log dir in disk persistence mode");
+                }
+                cmd.append(" -l ").append(params.getLogDir());
+                break;
+            case MEMORY:
+                cmd.append(" -m");
+                break;
+        }
+
+        if (params.getMode() == Mode.SINGLE) {
+            cmd.append(" -s");
+        }
+
+        cmd.append(" -d ").append(params.getLogLevel().toString()).append(" ");
+
+        cmd.append(params.getPort());
+
+        String cmdLineParams = cmd.toString();
+        log.trace("Command line parameters: {}", cmdLineParams);
+
+        return cmdLineParams;
+    }
+
+    /**
+     * Print container logs.
+     *
+     * @param containerId docker container id
+     * @throws Exception docker exception
+     */
+    private void logs(String containerId) throws DockerException, InterruptedException {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused private "logs" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AUnusedPrivateMethod)